blog/2024-01-09-introducing-bidirectional-github-sync/index.mdx

```diff
@@ -0,0 +1,69 @@
+
+Studio also provides a number of other features like test generation and export options to code, markdown, or to external editors. But this work can all be in coordination with the underlying GitHub repo.
+
+### Staying in sync with the GitHub repo
+
+By default, connected projects will [automatically be synchronized with the GitHub repo](https://stately.ai/docs/import-from-github#sync-changes-to-machines-from-github) every time the project loads. New files added to selected folders in the repository will appear in Studio. We’ll also warn of any deleted files that might cause machines to be deleted in Studio. It’s possible to opt out of this and to perform syncs manually but we recommend that this still be done fairly often to prevent potential conflicts.
+
+![By default, connected repos are auto-synced with the GitHub repo but this can be disabled if you want to manually sync instead.](auto-sync.png)
+
+A connected project starts out [in sync with the branch](https://stately.ai/docs/import-from-github#sync-changes-to-machines-from-github) that was specified during the import process via the project's [auto-sync setting](https://stately.ai/docs/import-from-github#auto-sync). However, it is also possible to [connect to additional repo branches](https://stately.ai/docs/import-from-github#sync-changes-to-machines-from-github). Under the hood, these branch connections are considered a different version of the connected project and it’s easy to switch between them, add new branches, or delete branch connections.
+
+At any time, new machines can be added directly to a connected project or existing machines can be moved or forked into a connected project. Those can all be [added to the GitHub repo](https://stately.ai/docs/import-from-github#sync-new-machines-created-in-stately-studio) as part of the pull request process.
+
+### Creating pull requests back to the repo
+
+The key feature required to complete a bidirectional workflow between GitHub and Stately Studio is the ability to [create pull requests](https://stately.ai/docs/import-from-github#create-pull-request) back to the original repository and that’s now easy to do directly from Studio. Connected projects have a GitHub Settings panel where you can manage the GitHub connection to your project, including creating a pull request which can target any other branch in the repository.
+
+![Connected projects have a GitHub Settings panel where users can create new pull requests targeting any branch in the connected repository.](pull-request-form.png)
+
+Once a pull request is made, the process becomes the same as it is for any other GitHub pull request where it can be reviewed, modified, approved, and merged. Subsequent changes can be added to existing pull requests with the [Update pull request feature](https://stately.ai/docs/import-from-github#update-pull-request). This allows for greater productivity in Studio while knowing that changes will make their way back into the GitHub repo.
```

Comment:
```suggestion
Once a pull request is made, the process becomes the same as it is for any other GitHub pull request where it can be reviewed, modified, approved, and merged. Subsequent changes can be added to existing pull requests with the [Update pull request feature](https://stately.ai/docs/import-from-github#update-pull-request). This allows for greater productivity in the Studio while knowing that changes will make their way back into the GitHub repo.
```